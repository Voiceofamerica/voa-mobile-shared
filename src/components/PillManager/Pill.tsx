
import * as React from 'react'

import { themed, ThemeProps, DEFAULT_THEME } from '../ThemeProvider'
import { pill } from './PillManager.scss'

import {
  divider,
} from './PillManager.scss'

export interface Props extends ThemeProps {
  selected?: boolean
  onClick?: () => void
  className?: string
  style?: React.CSSProperties
}

@themed
class Pill extends React.Component<Props> {
  render () {
    const {
      selected = false,
      onClick = () => null,
      className = '',
      style = {},
      theme = DEFAULT_THEME,
      children,
    } = this.props

    const {
      pillBackground,
      pillColor,
      pillSelectedBackground,
      pillSelectedColor,
    } = theme

    const color = selected ? pillSelectedColor : pillColor
    const background = selected ? pillSelectedBackground : pillBackground

    return (
      <div className={`${pill} ${className}`} style={{ color, background, ...style}} onClick={() => onClick()}>
        {children}
      </div>
    )
  }
}

export default Pill
