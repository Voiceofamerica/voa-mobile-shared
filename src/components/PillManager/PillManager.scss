
.pillManager {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  flex-wrap: wrap;

  .pill {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-grow: 0;
    flex-shrink: 0;
    min-width: 16vw;
    height: 20px;
    font-size: 14px;
    margin-top: 6px;
    padding: 4px 3vw;
    border-radius: 20px;
  }

  .divider {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    width: 100%;
    margin-top: 6px;
  }
}
