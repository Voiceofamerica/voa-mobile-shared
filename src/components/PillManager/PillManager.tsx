
import * as React from 'react'

import {
  pillManager,
} from './PillManager.scss'

export interface Props {
  className?: string
  style?: React.CSSProperties
}

class PillManager extends React.Component<Props> {
  render () {
    const { className = '', style = {}, children } = this.props

    return (
      <div className={`${pillManager} ${className}`} style={style}>
        {children}
      </div>
    )
  }
}

export default PillManager
