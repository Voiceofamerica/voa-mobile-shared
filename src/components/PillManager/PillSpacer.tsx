
import * as React from 'react'

import {
  divider,
} from './PillManager.scss'

export interface Props {
  className?: string
  style?: React.CSSProperties
}

class PillSpacer extends React.Component<Props> {
  render () {
    const { className = '', style = {}, children } = this.props

    return (
      <div className={`${divider} ${className}`} style={style}>
        {children}
      </div>
    )
  }
}

export default PillSpacer
