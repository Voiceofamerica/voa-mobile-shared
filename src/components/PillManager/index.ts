
export { default } from './PillManager'
export { default as Pill } from './Pill'
export { default as PillSpacer } from './PillSpacer'
