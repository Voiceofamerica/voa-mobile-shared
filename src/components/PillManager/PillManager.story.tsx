
import * as React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

import ThemeProvider, { BaseTheme, PopupButtonGroupTheme, DEFAULT_THEME } from '../ThemeProvider'
import SvgIcon from '../SvgIcon'

import PillManager from './PillManager'
import Pill from './Pill'
import PillSpacer from './PillSpacer'

interface DisplayState {
  top: number
  show: boolean
}

const testBaseTheme: BaseTheme = {
  lightGrey: '#666666',
  black: '#DDDDFF',
  grey: '#AAAAAA',
}

const testLocalTheme: PopupButtonGroupTheme = {
  popupButtonBackground: '#003300',
  popupButtonColor: '#DDFFDD',
  popupButtonBorder: '#664499',
}

const startContent = [
  'first',
  'second',
  'third',
  'fourth',
  'fifth',
  'this is a particularly long pill that will more than likely break the flow',
  'sixth',
  'seventh',
]

const endContent = [
  'eighth',
  'ninth',
  'tenth',
  'eleventh',
  'twelfth',
  'thirteenth',
  'fourteenth',
  'fifteenth',
]

const clickAction = action('pill clicked')

storiesOf(PillManager.name, module)
  .add('default display', () => (
    <PillManager>
      {
        startContent.map((val, idx) => (
          <Pill key={val} selected={idx === 4} onClick={() => clickAction('start', val)}>{val}</Pill>
        ))
      }
      <PillSpacer>
        <div>
          <div style={{ fontSize: '1.3em', fontWeight: 'bold' }}>This is a spacer</div>
          It takes the whole width
        </div>
        <div>And it's flex</div>
      </PillSpacer>
      {
        endContent.map((val) => (
          <Pill key={val} onClick={() => clickAction('end', val)}>{val}</Pill>
        ))
      }
    </PillManager>
  ))
