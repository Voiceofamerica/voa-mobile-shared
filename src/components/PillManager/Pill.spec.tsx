
import * as React from 'react'
import { create } from 'react-test-renderer'
import { mount } from 'enzyme'

import Pill from './Pill'
import { pill } from './PillManager.scss'

describe(`<${Pill.name} />`, () => {
  describe('shapshots', () => {
    it('should render without children', () => {
      const element = create((
        <Pill />
      ))

      expect(element).toMatchSnapshot()
    })
    it('should render with children', () => {
      const element = create((
        <Pill>
          This should show up
        </Pill>
      ))

      expect(element).toMatchSnapshot()
    })
    it('should render selected', () => {
      const element = create((
        <Pill selected />
      ))

      expect(element).toMatchSnapshot()
    })
    it('should render with className', () => {
      const element = create((
        <Pill className='myClass' />
      ))

      expect(element).toMatchSnapshot()
    })
    it('should render with style', () => {
      const element = create((
        <Pill style={{ background: 'myColor' }} />
      ))

      expect(element).toMatchSnapshot()
    })
  })

  describe('onClick', () => {
    it('should be called whenever the pill is pressed', () => {
      const onClick = jest.fn()

      const tree = mount((
        <Pill onClick={onClick} />
      ))

      tree.find(`.${pill}`).simulate('click')

      expect(onClick).toHaveBeenCalledTimes(1)
    })
  })
})
