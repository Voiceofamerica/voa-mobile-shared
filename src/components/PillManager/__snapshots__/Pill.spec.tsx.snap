// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pill /> shapshots should render selected 1`] = `
<div
  className="pill "
  onClick={[Function]}
  style={
    Object {
      "background": "#1AB088",
      "color": "#FFFFFF",
    }
  }
/>
`;

exports[`<Pill /> shapshots should render with children 1`] = `
<div
  className="pill "
  onClick={[Function]}
  style={
    Object {
      "background": "#0061B0",
      "color": "#FFFFFF",
    }
  }
>
  This should show up
</div>
`;

exports[`<Pill /> shapshots should render with className 1`] = `
<div
  className="pill myClass"
  onClick={[Function]}
  style={
    Object {
      "background": "#0061B0",
      "color": "#FFFFFF",
    }
  }
/>
`;

exports[`<Pill /> shapshots should render with style 1`] = `
<div
  className="pill "
  onClick={[Function]}
  style={
    Object {
      "background": "myColor",
      "color": "#FFFFFF",
    }
  }
/>
`;

exports[`<Pill /> shapshots should render without children 1`] = `
<div
  className="pill "
  onClick={[Function]}
  style={
    Object {
      "background": "#0061B0",
      "color": "#FFFFFF",
    }
  }
/>
`;
